proposals/stdlib/random.md

```diff
@@ -0,0 +1,245 @@
+* Random game content and level generation
+* Simulations requiring a source of random numbers
+* Using randomness in tests to cover more code execution paths
+
+## Alternatives
+
+The current alternatives require finding some platform libraries and providing common multiplatform API
+for them or implementing random generators by hand. 
+
+## Placement
+
+This API shall be placed into the Kotlin Standard Library. Regarding the package the following options were considered:
+
+- Use the existing `kotlin` package which is imported by default.
+- Use the existing `kotlin.math` package
+- Create new `kotlin.random` package and place API there.
+
+We decided to stick with the latter as it would be more explorable in docs: all the related API will be shown close to each other.
+Also it's more future proof in regard to possible API additions.
+
+A downside of a new package is that it will require new imports, especially if there will be extensions.
+A class named `Random` from that package can be confused with `java.util.Random` in the completion list.
+It should be investigated whether we could make that package imported by default in Kotlin 1.3.
+
+## Reference implementation
+
+The implementation will be available for preview in 1.3-M1.
+
+## Dependencies
+
+What are the dependencies of the proposed API:
+
+* JDK: `java.util.Random` in JDK 6,7, `ThreadLocalRandom` in JDK 8+
+* JS: `Math.random` function as the source of randomness seed
+
+## API details
+
+The following code snippet summarizes the public API introduced by this proposal.
+
+```kotlin
+abstract class Random {
+
+    abstract fun nextBits(bitCount: Int): Int
+
+    open fun nextInt(): Int 
+    open fun nextInt(bound: Int): Int 
+    open fun nextInt(origin: Int, bound: Int): Int
+    open fun nextInt(range: IntRange): Int
+
+    open fun nextLong(): Long
+    open fun nextLong(bound: Long): Long
+    open fun nextLong(origin: Long, bound: Long): Long
+    open fun nextLong(range: LongRange): Long
+    
+    open fun nextBoolean(): Boolean
+   
+    open fun nextDouble(): Double
+    open fun nextDouble(bound: Double): Double
+    open fun nextDouble(origin: Double, bound: Double): Double
+    open fun nextFloat(): Float
+    
+    open fun nextBytes(array: ByteArray, fromIndex: Int = 0, toIndex: Int = array.size): ByteArray
+    open fun nextBytes(array: ByteArray): ByteArray
+    open fun nextBytes(size: Int): ByteArray 
+
+    // The default random number generator
+    // Safe to use concurrently in JVM, and to transfer between workers in Native    
+    companion object : Random() {
+        // overrides all methods delegating them to some platform-specific RNG implementation
+    }
+}
+
+// constructor-like functions to get a reproducible RNG implementation with the specified seed
+fun Random(seed: Int): Random
+fun Random(seed: Long): Random
+
+// JVM-only: functions to wrap java.util.Random in kotlin.random.Random and vice-versa
+fun java.util.Random.asKotlinRandom(): Random
```

Comment:
No, because "to" would suggest that you create a new random instance which is no longer dependent on the original. This only wraps the java random instance. 
This is consistent with the naming. One other example of this would be [`List<T>.asReversed`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/as-reversed.html).

Reply:
I personally dont understand why we need `Random.asJavaRandom()` and `java.util.Random.asKotlinRandom()` on the Random object. Anyone can explain it to me ?

Reply:
Thanks for your explanation, it's more clear for me now. If tomorrow, the implementation of ThreadLocalRandom change in Java, these methods will be a problem no ? (because of multiplatform compatibility constrain)

Reply:
There's no multiplatform compatibility constraint for an arbitrary implementation of `Random`, only for that returned by `Random(seed)` function.